### PR TITLE
Add model management methods to Natural Language Understanding

### PR DESCRIPTION
**Pull request by g-may**
Successfully listed a custom model from my own service instance and deleted it with these methods.

### CODE REVIEW

File: watson_developer_cloud/natural_language_understanding_v1.py [hunk 1]
```diff
@@ -82,3 +82,24 @@ def analyze(self, features, text=None, url=None, html=None,
...
+        """
+        return self.request(method='GET', url='/v1/models',
+                            params={"version": self.version},
+                            accept_json=True)
+
+    def deleteModel(self, model_id=None):
+        """
+        Deletes a custom model
+        :param model_id: The ID of the model to delete
+        :return: dict of analyzed text
```
Comment (kognate) [inline watson_developer_cloud/natural_language_understanding_v1.py]: Model deletion really runs a dict of analyzed text?

File: watson_developer_cloud/natural_language_understanding_v1.py [hunk 2]
```diff
@@ -82,3 +82,24 @@ def analyze(self, features, text=None, url=None, html=None,
...
+    def listModels(self):
+        """
+        Lists the custom models available for your service instance
+        :return: dict of available custom models
+        """
+        return self.request(method='GET', url='/v1/models',
+                            params={"version": self.version},
+                            accept_json=True)
+
+    def deleteModel(self, model_id=None):
```
Comment (kognate) [inline watson_developer_cloud/natural_language_understanding_v1.py]: You indicate that model_id can be None by making it's default value none.  I'd remove the default value and the model_id check later.    The service call will error on the None and by making it required the users knows that it should be there.

File: watson_developer_cloud/natural_language_understanding_v1.py (hunk 1 above)
Comment (g-may) [inline watson_developer_cloud/natural_language_understanding_v1.py]: sure doesn't. thanks

File: watson_developer_cloud/natural_language_understanding_v1.py [hunk 3]
```diff
@@ -82,3 +82,24 @@ def analyze(self, features, text=None, url=None, html=None,
...
+        """
+        Deletes a custom model
+        :param model_id: The ID of the model to delete
+        :return: dict with status of model deletion
+        """
+        if model_id is None:
+            raise ValueError("Missing parameter 'model_id'")
+
+        return self.request(method='DELETE', url='/v1/models/'+model_id,
```
Comment (kognate) [inline watson_developer_cloud/natural_language_understanding_v1.py]: Format strings   ` '/v1/models/{0}'.format(model_id)` are preferred over string concatenation.

File: watson_developer_cloud/natural_language_understanding_v1.py (hunk 2 above)
Comment (g-may) [inline watson_developer_cloud/natural_language_understanding_v1.py]: I was just mirroring how the features parameter is checked by the SDK in the analyze endpoint. Does the new commit fix things?

File: watson_developer_cloud/natural_language_understanding_v1.py (hunk 3 above)
Comment (g-may) [inline watson_developer_cloud/natural_language_understanding_v1.py]: fixed, thanks

**Comment by kognate** [comment]
It does.

The difference is that params are optional in the analye method.  One of
them is required, and they are mutually exclusive.   In the listmodels
method the model_id is really required and isn't part of a set.

**Pull request merged**